### PR TITLE
feat(annotations): add EcotReasoningModule for dense chain-of-thought supervision

### PR DESCRIPTION
**Pull request by smellslikeml**
## Summary / Motivation

Adapts the `lerobot-annotate` pipeline to produce Dense Embodied Chain-of-Thought (ECoT) reasoning supervision alongside the existing plan/vqa/interjection annotations. Instead of only subtask labels, the dataset now stores a sequence of intermediate reasoning steps (scene perception, object identification, task plan, subtask decomposition) per trajectory segment.

This is a contract-anchored addition to the existing steerable pipeline: a new module (`EcotReasoningModule`) with the same `episode + FrameProvider + VlmClient → staged-column` I/O shape as the plan/vqa modules, reusing the Qwen-VL family the repo already runs and the ECoT schema already referenced in `prompts/vqa.txt` (Zawalski 2024). A real run of the pipeline emits the traces via the existing executor → validator → writer path into the `language_persistent` parquet column; the module is also exported as public library API from `modules/__init__.py`.

Adapted from ZR-0 (arxiv:2606.30552) — the ECoT trace schema and dense-reasoning supervision, ported to lerobot's annotation surface.

_Per [AI_POLICY.md](https://github.com/huggingface/lerobot/blob/main/AI_POLICY.md): initial draft assisted by [Outrider](https://github.com/remyxai/outrider), then reviewed, tested against `lerobot/aloha_static_coffee` end-to-end, and iterated by @smellslikeml before filing._

**Out of scope** (the repo hosts the supervision, not the policy that trains on it):
- ZR-0's dual-stream VLA architecture (pre-trained VLM System 2 + Diffusion Transformer action expert System 1) and its cross-attention coupling — a 2.6B trained policy the repo does not host.
- Flow-matching action-chunk generation and the action expert — policy-training concerns, out of scope for an annotation module.
- The inference-time ECoT-skip attention mask — an architecture/training detail, not an annotation output.
- ProcCorpus-60M pre-training (~60M frames / ~1,000h / 400k+ trajectories) — the paper's training corpus; not reproduced.
- No VLA training loop, no model weights, and no benchmark eval (LIBERO / RoboTwin 2.0 / RoboCasa GR-1 / xArm).

## Related issues
- Related: #3132 (subtask-only auto-annotation via a separate `subtasks.parquet` table — adjacent scope but distinct: ECoT emits a 4-field joint reasoning trace per anchor into the existing `language_persistent` column, not a subtask segmentation).
- Fixes / Closes: # N/A

## What changed

- `src/lerobot/scripts/lerobot_annotate.py` — `annotate()` (entry point `lerobot-annotate=lerobot.scripts.lerobot_annotate:main`, registered in `pyproject.toml` `[project.scripts]`) now constructs `EcotReasoningModule` and passes it to the Executor.
- `src/lerobot/annotations/steerable_pipeline/modules/` — new `EcotReasoningModule`: anchors a timestamp at a configurable cadence (default 0.5 Hz), grounds a contact-sheet window of frames around it, issues one structured-JSON VLM call per anchor, and coerces/partial-tolerates the canonical 4-field ECoT schema; exported as public API from `modules/__init__.py`.
- `src/lerobot/datasets/language.py` — `'ecot'` style registered as core + persistent + camera-agnostic, so the existing `StagingValidator` and `LanguageColumnsWriter` land the traces in `language_persistent` with one row active from its anchor until the next (mirroring plan/memory rows).
- `src/lerobot/annotations/steerable_pipeline/` (Executor) — wired in as phase 4.5, skipped when the module is disabled or not attached.

## How was this tested (or how to run locally)

**Unit tests** — 3/3 pass in `tests/annotations/test_ecot_reasoning.py`:

```bash
uv sync --extra dataset
uv pip install draccus pytest pytest-timeout pytest-cov
uv run pytest tests/annotations/test_ecot_reasoning.py -v
```

**End-to-end** — smoke-tested against `lerobot/aloha_static_coffee` episode 0 with a small Qwen2.5-VL-based VLM (3B, Q4_K_M GGUF served via `llama-server` on CPU). The `openai` VlmClient backend talks to any OpenAI-compatible endpoint, so no cloud API key required. Produced valid 4-field ECoT traces; the client's `_extract_first_json_object` fallback handles reasoning-style VLMs that emit `<think>...</think>` before the answer.

## Checklist (required before merge)
- [x] Linting/formatting run (`pre-commit run -a`) — full sweep passes: ruff, ruff-format, prettier, typos, pyupgrade, gitleaks, zizmor, bandit, mypy all green
- [x] All tests pass locally (`pytest`) — 3/3 in `tests/annotations/test_ecot_reasoning.py`
- [x] Documentation updated — `docs/source/annotation_pipeline.mdx` updated with ECoT module in the intro + ASCII diagram
- [x] CI is green — will re-verify on upstream after filing
- [x] Community Review: I have reviewed another contributor's open PR and linked it here: huggingface/lerobot#3896

## Reviewer notes
- The ZR-0 reference implementation ships Apache-2.0-licensed code backing the ECoT schema and trace format ([github.com/RUCKBReasoning/ZR-0](https://github.com/RUCKBReasoning/ZR-0)).
- **Relationship to the `plan` module**: `scene_perception` and `object_identification` are new perceptual output fields not emitted by the plan module. The overlap on `task_plan` / `subtask_decomposition` is intentional — the ECoT trace is a joint dense-reasoning artifact per anchor (matching the Zawalski 2024 / ZR-0 training-data recipe), not a plan decomposition. Both modules can be enabled together without duplication in `language_persistent` since the styles are distinct (`plan` / `memory` / `ecot`).

Co-Authored-By: remyx-ai[bot] <289541483+remyx-ai[bot]@users.noreply.github.com>

### CODE REVIEW

**Comment by smellslikeml** [comment]
Thanks — all three items addressed in the amended commit:

1. **JSON key order** — dropped `sort_keys=True`; `content` now emits fields in the canonical `_ECOT_FIELDS` order (perception → identification → planning → decomposition). Added a serialized-order assertion to the test.
2. **Prompt loading** — moved the template to `prompts/ecot.txt` and load it via `load_prompt("ecot")` matching the other modules (also enables the `LEROBOT_PROMPT_OVERRIDE_ecot` env-var path).
3. **No-camera warning** — added a one-time warning + zero-emit path mirroring `GeneralVqaModule`'s pattern, with a new test that asserts the warning fires exactly once.

Verified end-to-end locally against `lerobot/aloha_static_coffee` episode 0 with a quantized Qwen2.5-VL-based VLM.

**Comment by smellslikeml** [comment]
@pkooij just merged upstream/main to clear the docs conflict — PR is now up-to-date and mergeable. Would appreciate another look whenever you have a moment.
